katcp/inspecting_client.py

```diff
@@ -757,20 +762,20 @@ def inspect_sensors(self, name=None, timeout=None):
         if not reply.reply_ok():
             # If an unknown sensor is specified the desired result is to return
             # an empty list, even though the request will fail
-            if name is None or 'Unknown sensor' not in reply.arguments[1]:
+            if name is None or b'Unknown sensor' not in reply.arguments[1]:
                 raise SyncError('Error reply during sync process: {}'
                                 .format(reply))
 
 
         sensors_old = set(self._sensors_index.keys())
         sensors_updated = set()
         for msg in informs:
-            sen_name = msg.arguments[0]
+            sen_name = ensure_native_str(msg.arguments[0])
             sensors_updated.add(sen_name)
-            sen = {'description': msg.arguments[1],
-                   'units': msg.arguments[2],
-                   'sensor_type': msg.arguments[3],
-                   'params': msg.arguments[4:]}
+            sen = {'description': ensure_native_str(msg.arguments[1]),
+                   'units': ensure_native_str(msg.arguments[2]),
+                   'sensor_type': ensure_native_str(msg.arguments[3]),
+                   'params': [ensure_native_str(arg) for arg in msg.arguments[4:]]}
```

Comment:
Elsewhere in the code the parameters (other than name) are kept as `bytes` (e.g. L971). Would it be better to keep things consistent?